.github/workflows/ci.yml

```
@@ -0,0 +1,459 @@
+        uses: shivammathur/setup-php@6cba851606e69e00775ebccba68bc1d418b6fa44 # @v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: curl, dom, gd, intl, json, ldap, mbstring, mysql, tidy, xdebug, zip
+          tools: composer:v2
+          coverage: xdebug
+          # While this should be the correct way to allow forks in composer.json repositories
+          # in practice there are still many sporadic "Could not authenticate against github.com" errors
+          # there's 1,000 requests per hour limit when using this token, likely it get exceeded
+          # fairly easily when using a fork with multiple jobs in a matrix
+          #env:
+          # COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Configure PHP
+        run: |
+          # Set memory limit and disable xdebug if not running phpcoverage
+          if [ -z $(which php) ]; then
+            echo "PHP not installed, skipping" && exit 0
+          fi
+
+          # github linux runners have 7GB of RAM
+          # https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+          # Set a high memory limit, particularly for php coverage tests
+          PHP_MEMORY_LIMIT=6144M
+          # Assign less memory for behat tests so that chrome has plenty of memory available
+          if [ "${{ matrix.endtoend }}" == "true" ]; then
+            PHP_MEMORY_LIMIT=4096M
+          fi
+          echo "PHP_MEMORY_LIMIT is $PHP_MEMORY_LIMIT"
+
+          sudo sh -c "echo 'memory_limit = $PHP_MEMORY_LIMIT' >> /etc/php/${{ matrix.php }}/cli/php.ini"
+          if [ -f /etc/php/${{ matrix.php }}/apache2/php.ini ]; then
+            sudo sh -c "echo 'memory_limit = $PHP_MEMORY_LIMIT' >> /etc/php/${{ matrix.php }}/apache2/php.ini"
+          fi
+
+          # Disable xdebug which greatly slow down unit testing
+          # Note: omitting xdebug from shivammathur/setup-php still results in xdebug being installed and enabled
+          if [ "${{ matrix.phpcoverage }}" == "true" ]; then
+            sudo sh -c "echo ';zend_extension=xdebug.so' > /etc/php/${{ matrix.php }}/mods-available/xdebug.ini"
+          fi
+          echo "PHP has been configured"
+
+      - name: Apt install additional requirements 
+        run: |
+          # apt install extra requirements as required
+          if [ "${{ matrix.endtoend }}" == "true" ]; then
+            sudo apt install -y software-properties-common
+            sudo add-apt-repository -y ppa:ondrej/php
+            sudo add-apt-repository -y ppa:ondrej/apache2
+            sudo apt update
+            sudo apt install -y libapache2-mod-php${{ matrix.php }}
+            # ubuntu-latest comes with a current version of google-chrome-stable and chromedriver
+          fi
+          if [[ "${{ github.repository }}" =~ /(spellcheck|recipe-authoring-tools)$ ]] || [ "${{ matrix.phpunit_suite }}" == "recipe-authoring-tools" ]; then
```

```suggestion
          if [[ "${{ github.repository }}" =~ ^silverstripe/(silverstripe-spellcheck|recipe-authoring-tools)$ ]] || [ "${{ matrix.phpunit_suite }}" == "recipe-authoring-tools" ]; then
```
Unless there's a reason to not specify the silverstripe organisation here?

---

If we're doing dev on creative-commoners or a personal account then we don't want the account to be part of the prefix

---

I'm trying to understand in what situation we'll be running CI on creative-commoners... I suppose it ultimately doesn't matter too much, the chance of someone having a module named the same as one of these isn't too likely.